lib/internal/util/debuglog.js

```diff
@@ -17,20 +16,19 @@ const { inspect, format, formatWithOptions } = require('internal/util/inspect');
 // debuglog() before initializeDebugEnv() is called will throw.
 let debugImpls;
 
-let debugEnvRegex = /^$/;
-let testEnabled;
+let testEnabled = () => false;
 
 // `debugEnv` is initial value of process.env.NODE_DEBUG
 function initializeDebugEnv(debugEnv) {
   debugImpls = ObjectCreate(null);
   if (debugEnv) {
+    // This is run before any user code, it's OK not to use primordials.
     debugEnv = debugEnv.replace(/[|\\{}()[\]^$+?.]/g, '\\$&')
-      .replace(/\*/g, '.*')
-      .replace(/,/g, '$|^')
-      .toUpperCase();
-    debugEnvRegex = new RegExp(`^${debugEnv}$`, 'i');
+      .replaceAll('*', '.*')
+      .replaceAll(',', '$|^');
```

Comment:
```suggestion
      .replaceAll(',', '$|^')
      .toUpperCase();
```
Shouldn't we keep the `toUpperCase`?

Reply:
I don't think it's useful, the regex uses the `i` flag (case insensitivity).